scripts/amtoine-upl

```diff
@@ -58,26 +58,30 @@ def pipx_deps [] {
 }
 
 
-let x = {
-    pkgs: {
-        pacman:
-            {
-                all: (pacman_deps_wrapper --flag "-Q"),
-                explicit: (pacman_deps_wrapper --flag "-Qe"),
-                native: (pacman_deps_wrapper --flag "-Qen"),
-                foreign: (pacman_deps_wrapper --flag "-Qem")
-            }
-        rust:
-            {
-                cargo: (cargo_deps)
-            }
-        python:
-            {
-                pip: (pip_deps),
-                pipx: (pipx_deps)
-            }
+def main [
+    output: string = "~/pkgs.toml"
```

Comment:
Is it a default arg?
Is so where is $output defined?

Reply:
a few things about `nushell` scripts, and let's say the script is called `foo.nu`:
- `./foo.nu` will call what's inside the `main` function
- the signature of an argument is as follows: `name: type = default`
  - if `default` is specified, the `name` argument is not required in the `./foo.nu` call
  - if `default` is not specified, the `name` argument is mandatory

so what does it do here? :thinking: 

- calling `./scripts/amtoine-upl` will put the `pkgs.toml` file in `~`
- calling `./scripts/amtoine-upl /path/to/somewhere` will put the `pkgs.toml file in `/path/to/somewhere/`
 
:yum: 

in the end, `$output` comes from the CLI arguments :+1:

Reply:
as agreed in private, i'll resolve this thread :relieved: